### PR TITLE
Add/edit check windows.

### PR DESCRIPTION
**Pull request by pris54**
Stories 15007 and 15097

### CODE REVIEW

**Comment by pris54** [comment]
@activemq could you please look at the error thrown by Travis?

File: admin/models/check-window.js [hunk 1]
```diff
@@ -11,28 +11,30 @@ const CheckWindow = new Schema({
...
     }]
   }
 }, {timestamps: true})
 
+/**
+ * Retrieve all check windows and sort by arguments passed.
+ * @param sortBy
+ * @param direction
+ * @returns {Promise}
+ */
```
Comment (jon-shipley) [inline admin/models/check-window.js]: Good!

File: admin/routes/index.js [hunk 2]
```diff
@@ -41,60 +39,7 @@ router.get('/profile', isAuthenticated(), (req, res) => getProfile(req, res))
 /* Administration page */
 router.get('/administrator', isAuthenticated(config.ROLE_TEST_DEVELOPER), (req, res) => getAdministration(req, res))
 /* Health check */
-async function getPing (req, res) {
```
Comment (jon-shipley) [inline admin/routes/index.js]: We seem to have lost some functionality here.`getPing()` should restore commitId and build number now.

File: admin/views/administrator/check-windows-form.ejs [hunk 3]
```diff
@@ -0,0 +1,199 @@
...
+                </label>
+                <div class="form-date">
+                    <div class="form-group form-group-day">
+                        <label class="form-label" for="checkStartDay">Day</label>
+                        <input class="form-control <%= error.checkStartDay ? 'form-control-error' : '' %>"
+                               id="checkStartDay"
+                               name="checkStartDay"
+                               type="number"
+                               value="<%= checkWindowData.checkStartDay %>"
+                               aria-required="true"
```
Comment (jon-shipley) [inline admin/views/administrator/check-windows-form.ejs]: Good to see the aria-* attributes used here. Perhaps you could come up with documentation to standardise this (in another PR).

File: admin/controllers/administrator.js [hunk 4]
```diff
@@ -195,9 +198,137 @@ const getCheckWindows = async (req, res, next) => {
...
+      errorMessage: checkWindowErrorMessages,
+      actionName,
+      urlActionName,
+      breadcrumbs: req.breadcrumbs()
+    })
+  }
+
+  if (req.body.checkWindowId !== '') {
+    try {
+      checkWindow = await CheckWindow.findOne({_id: req.body.checkWindowId}).exec()
```
[the remark this answers is not in the document]
Comment (pris54) [reply admin/controllers/administrator.js]: @jon-shipley that's true however, if there is a human error during editing, the id will come for sure from the hidden field

File: admin/views/administrator/check-windows-form.ejs (hunk 3 above)
Comment (pris54) [reply admin/views/administrator/check-windows-form.ejs]: @jon-shipley Good idea, I'll bring the need for a document during sprint planning.

File: admin/routes/index.js (hunk 2 above)
Comment (pris54) [reply admin/routes/index.js]: @jon-shipley Odd... Code restored.

File: admin/controllers/administrator.js [hunk 5]
```diff
@@ -195,9 +198,137 @@ const getCheckWindows = async (req, res, next) => {
...
+ */
+const checkWindowsForm = async (req, res, next) => {
+
+  let error = ''
+  let errorMessage = ''
+  let checkWindowData = ''
+  let successfulPost = false
+  let actionName = (req.params.action === 'add' && req.params.id === undefined) ? 'Create' : 'Edit'
+  let urlActionName = req.params.action
+  let currentYear = Date.now()
```
Comment (jon-shipley) [inline admin/controllers/administrator.js]: This is not exactly the current year...

File: admin/controllers/administrator.js (hunk 4 above)
[the remark this answers is not in the document]
Comment (jon-shipley) [reply admin/controllers/administrator.js]: I'm not fussed about the param - but in one you use `findOne` and in the other you use `CheckWindow.getCheckWindow`.  Seems inconsistent.  Are they doing the same thing?

File: admin/routes/index.js [hunk 6]
```diff
@@ -94,7 +93,7 @@ function getBuildNumber () {
   })
 }
 
-router.get('/ping', (req, res) => getPing(req, res))
+router.get('/ping', (req, res) => res.status(200).send('OK'))
```
Comment (jon-shipley) [inline admin/routes/index.js]: Still doesn't look right... should return the commit and and buildId

File: admin/controllers/administrator.js (hunk 5 above)
Comment (pris54) [reply admin/controllers/administrator.js]: Fixed, please have a look

File: admin/routes/index.js (hunk 6 above)
Comment (pris54) [reply admin/routes/index.js]: This has to be a bad merge... I've fixed it.

File: admin/controllers/administrator.js (hunk 4 above)
[the remark this answers is not in the document]
Comment (pris54) [reply admin/controllers/administrator.js]: @jon-shipley yes, you're right, getCheckWindow was created later on. I've replace line 290 to use this method too.

**Pull request merged**